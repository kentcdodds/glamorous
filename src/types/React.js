// @flow
import type {Component, Element} from 'react'

export type FunctionalComponent = (
  {[key: string]: any}, // props
  ?{[key: string]: any}, // context
) => Element<*> | null;

// eslint-disable-next-line no-undef
export type ComponentClass = Class<Component<*, *, *>>;

export type RefCallback<Instance> = (Instance | null) => void;
