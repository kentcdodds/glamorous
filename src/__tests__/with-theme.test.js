/* eslint func-style:0 */
import React from 'react'
import {render} from 'enzyme'
import * as jestGlamorReact from 'jest-glamor-react'
import glamorous from '../'
import ThemeProvider from '../theme-provider'
import withTheme from '../with-theme'

expect.extend(jestGlamorReact.matcher)
expect.addSnapshotSerializer(jestGlamorReact.serializer)

test('renders a component with theme in props', () => {
  const Comp = withTheme(({theme}) => {
    return <glamorous.Div padding={theme.padding} />
  })
  expect(
    render(
      <ThemeProvider theme={{padding: '10px'}}>
        <Comp />
      </ThemeProvider>,
    ),
  ).toMatchSnapshotWithGlamor()
})
