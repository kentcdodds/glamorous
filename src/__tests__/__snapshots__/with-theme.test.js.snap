// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a component with theme in props 1`] = `
.css-1kle2qo,
[data-css-1kle2qo] {
  padding: 10px;
}

<div
  class="css-1kle2qo"
/>
`;
